python/cudf/cudf/_lib/csv.pyx

```diff
@@ -343,3 +351,49 @@ def read_csv(
             df = df.set_index(index_col)
 
     return df
+
+
+cpdef write_csv(
+    Table table,
+    file_path=None,
+    sep=",",
+    na_rep="",
+    header=True,
+    line_terminator="\n",
+    rows_per_chunk=8,
+
+):
+    """
+    Cython function to call into libcudf API, see `write_csv`.
+
+    See Also
+    --------
+    cudf.io.csv.write_csv
+    """
+
+    # Index already been reset and added as main column, so just data_view
+    cdef table_view input_table_view = table.data_view()
+    cdef bool include_header_c = header
+    cdef char delim_c = ord(sep)
+    cdef string line_term_c = line_terminator.encode()
+    cdef string na_c = na_rep.encode()
+    cdef int rows_per_hunk_c = rows_per_chunk
+    cdef table_metadata metadata_ = table_metadata()
+    cdef string true_value_c = 'True'.encode()
+    cdef string false_value_c = 'False'.encode()
+    file_path = str(os.path.expanduser(str(file_path))).encode()
+    cdef sink_info snk = sink_info(<string>file_path)
+
+    if header is True and table._column_names is not None:
+        metadata_.column_names.reserve(len(table._column_names))
+        for col_name in table._column_names:
+            metadata_.column_names.push_back(str.encode(col_name))
```

Comment:
Any reason you used `str.encode(col_name)` instead of `col_name.encode()` that is used elsewhere?

Reply:
Think I should have used str(col_name).encode, to cover `numerical` names as well